### ISSUE TITLE
Neural network in cpp

### ISSUE THREAD

**Issue by KunjShah95** [cneuralnetwork/Neural-Networks-in-every-Language#1]
#include <iostream>
#include <vector>
#include <cmath>

// Activation function (sigmoid)
double sigmoid(double x) {
    return 1 / (1 + exp(-x));
}

// Derivative of sigmoid
double sigmoid_derivative(double x) {
    return x * (1 - x);
}

int main() {
    // Training data (XOR gate)
    std::vector<std::vector<double>> training_inputs = {
        {0, 0},
        {0, 1},
        {1, 0},
        {1, 1}
    };
    std::vector<std::vector<double>> training_outputs = {
        {0},
        {1},
        {1},
        {0}
    };

    // Network architecture (2 input, 2 hidden, 1 output)
    int input_neurons = 2;
    int hidden_neurons = 2;
    int output_neurons = 1;

    // Weights and biases (randomly initialized)
    std::vector<std::vector<double>> hidden_weights(input_neurons, std::vector<double>(hidden_neurons));
    std::vector<double> hidden_biases(hidden_neurons);
    std::vector<std::vector<double>> output_weights(hidden_neurons, std::vector<double>(output_neurons));
    std::vector<double> output_biases(output_neurons);

    // Training loop
    int epochs = 10000;
    double learning_rate = 0.1;
    for (int epoch = 0; epoch < epochs; ++epoch) {
        for (int i = 0; i < training_inputs.size(); ++i) {
            // Forward pass
            std::vector<double> hidden_layer(hidden_neurons);
            for (int j = 0; j < hidden_neurons; ++j) {
                double weighted_sum = 0;
                for (int k = 0; k < input_neurons; ++k) {
                    weighted_sum += training_inputs[i][k] * hidden_weights[k][j];
                }
                hidden_layer[j] = sigmoid(weighted_sum + hidden_biases[j]);
            }

            std::vector<double> output_layer(output_neurons);
            for (int j = 0; j < output_neurons; ++j) {
                double weighted_sum = 0;
                for (int k = 0; k < hidden_neurons; ++k) {
                    weighted_sum += hidden_layer[k] * output_weights[k][j];
                }
                output_layer[j] = sigmoid(weighted_sum + output_biases[j]);
            }

            // Backpropagation
            std::vector<double> output_error(output_neurons);
            for (int j = 0; j < output_neurons; ++j) {
                output_error[j] = (training_outputs[i][j] - output_layer[j]) * sigmoid_derivative(output_layer[j]);
            }

            std::vector<double> hidden_error(hidden_neurons);
            for (int j = 0; j < hidden_neurons; ++j) {
                double error_sum = 0;
                for (int k = 0; k < output_neurons; ++k) {
                    error_sum += output_error[k] * output_weights[j][k];
                }
                hidden_error[j] = error_sum * sigmoid_derivative(hidden_layer[j]);
            }

            // Update weights and biases
            for (int j = 0; j < output_neurons; ++j) {
                for (int k = 0; k < hidden_neurons; ++k) {
                    output_weights[k][j] += learning_rate * output_error[j] * hidden_layer[k];
                }
                output_biases[j] += learning_rate * output_error[j];
            }

            for (int j = 0; j < hidden_neurons; ++j) {
                for (int k = 0; k < input_neurons; ++k) {
                    hidden_weights[k][j] += learning_rate * hidden_error[j] * training_inputs[i][k];
                }
                hidden_biases[j] += learning_rate * hidden_error[j];
            }
        }
    }

    // Test the network
    for (int i = 0; i < training_inputs.size(); ++i) {
        // ... (forward pass to get output_layer)
        std::cout << "Input: " << training_inputs[i][0] << " " << training_inputs[i][1] 
                  << ", Output: " << output_layer[0] << std::endl;
    }

    return 0;
}

**Comment by cneuralnetwork** [comment]
add this as a pull request I'll merge it

**Comment by KunjShah95** [comment]
done
